app/db/repository/authenticator_repository.rb

```diff
@@ -54,8 +54,17 @@ def load_authenticator(type:, account:, id:)
             args[variable.resource_id.split('/')[-1].underscore.to_sym] = variable.secret.value
           end
         end
+
         begin
-          @data_object.new(**args_list)
+          if Rails.configuration.feature_flags.enabled?(:pkce_support)
```

Comment:
Again, with testing in mind. I'd rather see either `configuration`, `feature_flags`,  or even `pkce_support` passed in as a dependency to the class, and use that when unit testing this, rather than setting the environment.

Reply:
Brillant. I didn't even think about that as an option 😢.  That may also allow us to remove the toggle in testing...